pkg/kubelet/kubelet_volumes.go

```diff
@@ -91,14 +91,18 @@ func (kl *Kubelet) newVolumeMounterFromPlugins(spec *volume.Spec, pod *v1.Pod, o
 
 // cleanupOrphanedPodDirs removes the volumes of pods that should not be
 // running and that have no containers running.  Note that we roll up logs here since it runs in the main loop.
-func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecontainer.Pod) error {
+func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecontainer.Pod, desiredPods map[types.UID]sets.Empty) error {
 	allPods := sets.NewString()
 	for _, pod := range pods {
 		allPods.Insert(string(pod.UID))
 	}
 	for _, pod := range runningPods {
 		allPods.Insert(string(pod.ID))
 	}
+	for uid := range desiredPods {
```

Comment:
I don't understand this change.

1) The code is to avoid cleanup pods in {pods || runningPods};
2) The new code is to avoid cleanup pods in {pods || runningPods || desiredPods}, which is a **superset** of 1), means that **we'll preserve more pod directories**;
3) The initial reason of calling `GetPods` again is to get latest `runningPods` set after killing unwanted pods, which makes the set smaller {pods || lessRunningPodsAfterKill}, which is a **subset** of 1).

If 3) is the reason of the extra `GetPods`, why do we remove it and do 2)?

Reply:
I got to #2 after fixing failing tests.

Let me see if I can get to #3 without breaking tests.